src/exceptions.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2024 Canonical Ltd.
```

Comment:
[Suggestion] You could have a `lib` folder and move this file and literals under it.  You may have some other candidates for this folder.

Reply:
Our charms in general work in this structure so I don't want to change it unless it becomes unmanageable.